astropy/time/core.py

```diff
@@ -1905,6 +1911,22 @@ def __gt__(self, other):
     def __ge__(self, other):
         return self._time_comparison(other, operator.ge)
 
+    def __array_ufunc__(self, ufunc, method, *args, **kwargs):
+        """Override numpy ufuncs. For now only isnan and isnat are implemented."""
+        if ufunc in (np.isnan, np.isnat):
+            return self._isnat()
+
+        if ufunc is np.isfinite:
+            return ~self._isnat()
+
+        return NotImplemented
```

Comment:
I fear this may break usage that currently goes through object arrays (say, `np.minimum(t1, t2)`), but haven't really checked.

Reply:
NotImplemented should allow the rest of the potential resolution mechanism to continue, right?

Reply:
But I see a lot of interactions of `mask` and the check if `jd2` is nan. So what's the actual design of the internal state supposed to be?

How do we differentiated "masked" from "has a nan value" in the internal state?

Reply:
> NotImplemented should allow the rest of the potential resolution mechanism to continue, right?

It signals that `Time` cannot handle it, but that means that the "default" mechanism of turning everything in an object array is by-passed. To be honest, that is for the better, since it is awfully slow (see example comparing straight addition to `np.add` below) and also generally one does not want object array output, but still, things that used to work will now no longer work. Anyway, more for later!

```
In [16]: from astropy.time import Time, TimeDelta

In [17]: t = Time(np.arange(50000, 50100), format='mjd')

In [18]: dt = TimeDelta(10, format='sec')

In [19]: %timeit t + dt
370 µs ± 1.57 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [20]: %timeit np.add(t, dt)
33.1 ms ± 102 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [21]: np.add(t, dt)
Out[21]: 
array([<Time object: scale='utc' format='mjd' value=50000.00011574074>,
       <Time object: scale='utc' format='mjd' value=50001.00011574074>,
       <Time object: scale='utc' format='mjd' value=50002.00011574074>,
       <Time object: scale='utc' format='mjd' value=50003.00011574074>,
.
.
.
```